### PR TITLE
[asset backfill] tweaks to avoid loading full graph

### PR DESCRIPTION
**Pull request by alangenfeld**
makes a few changes to facilitate loading only the relevant nodes instead of the whole graph to resolve information about an asset backfill 

## How I Tested These Changes

existing coverage

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#25736** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25736?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by OwenKephart — APPROVED** [review]
looks good, but seems like there was some linting stuff adding commas to a bunch of things which we might want to roll back

File: python_modules/dagster/dagster/_core/execution/asset_backfill.py [hunk 1]
```diff
@@ -331,7 +331,12 @@ def get_targeted_asset_keys_topological_order(
 
         Orders keys in the same topological level alphabetically.
         """
-        return [k for k in asset_graph.toposorted_asset_keys if k in self.target_subset.asset_keys]
+        nodes: List[BaseAssetNode] = [asset_graph.get(key) for key in self.target_subset.asset_keys]
+        return [
+            item
+            for items_by_level in toposort({node.key: node.parent_keys for node in nodes})
+            for item in sorted(items_by_level)
+        ]
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/execution/asset_backfill.py:339]: ?

File: python_modules/dagster/dagster/_core/execution/asset_backfill.py [hunk 2]
```diff
@@ -331,7 +331,12 @@ def get_targeted_asset_keys_topological_order(
 
         Orders keys in the same topological level alphabetically.
         """
-        return [k for k in asset_graph.toposorted_asset_keys if k in self.target_subset.asset_keys]
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/execution/asset_backfill.py:334]: X

**Pull request merged**